Author: David Kimdon <dwhedon@gordian.com>
Description: Specify which filename is causing an error if the filename is a
directory. (#45832)
--- a/bfd/opncls.c
+++ b/bfd/opncls.c
@@ -220,6 +220,13 @@ bfd_fopen (const char *filename, const c
 {
   bfd *nbfd;
   const bfd_target *target_vec;
+  struct stat s;
+
+  if (stat (filename, &s) == 0)
+    if (S_ISDIR(s.st_mode)) {
+      bfd_set_error (bfd_error_file_not_recognized);
+      return NULL;
+    }
 
   nbfd = _bfd_new_bfd ();
   if (nbfd == NULL)
